bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/misc/ExternalProgramImageDescriptor.java

```diff
@@ -62,29 +62,15 @@ public Image getImage() {
 		return createImage();
 	}
 
-	/**
-	 * Returns an SWT Image that is described by the information in this descriptor.
-	 * Each call returns a new Image.
-	 */
 	@Override
-	public ImageData getImageData() {
-		ImageData data = null;
-		ImageData defaultImage = WorkbenchImages.getImageDescriptor(ISharedImages.IMG_OBJ_FILE).getImageData();
-		if (defaultImage == null) {
-			return null;
+	public ImageData getImageData(int zoom) {
+		if (program != null) {
+			ImageData imageData = program.getImageData(zoom);
+			if (imageData != null) {
+				return imageData;
+			}
 		}
-
-		if (program == null || ((data = program.getImageData()) == null)) {
-			return defaultImage;
-		}
-
-		// The images in GNOME are too big. Scaling them does not give nice result so
-		// return defaultImage;
-		if (data.height > defaultImage.height || data.width > defaultImage.width) {
-			return defaultImage;
-		}
```

Comment:
I have verified that completely removing this fallback behavior for GNOME does not produce any regressions. My environment uses GNOME version 42.9.
Result: External program images look exactly the same with and without this code on GNOME. Here are example screenshots for 100% and 200% scaling, which are exactly the same with and without that code:
![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/755472/232ae7a9-1e27-4c25-9042-8bb70483d530)
![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/755472/03d4097d-06b2-4f42-a167-7adf67e62432)
Since the code has been introduced more than 20 yeras ago, it was probably some problem with GNOME environments back then.